FISH-230: Document automatic rest sslcontext (#261)
payara/Payara-Community-Documentation: docs/modules/ROOT/pages/documentation/ecosystem/payara-jax-rs-ssl.adoc

> = Payara JAX-RS Client Automatic SSL Context

Reviewer: Let us use the names (discussed also with marketing)

- JAX-RS extension -> for the page title and link on the ecosystems page.
- SSL Context Configuration -> for this feature

> JAX-RS Client Automatic SSL Context is an implementation to automatically indicate the certificate when doing a Rest Client call

Reviewer 2: This sentence is a bit wordy, and I'm not sure it actually describes what the feature is doing - it's not automatic.

To enable this we need to include the next dependency on the application

[source,XML]
----
<dependency>
  <groupId>com.ecosystem.extensions</groupId>
  <artifactId>ecosystem-rest-ssl-configuration</artifactId>
  <version>1.0</version>

> </dependency>

Reviewer 2: Artefact location obviosuly pending -  we don't own `com.ecosystem` coordinates.

----

Also it is needed to specify the next property for Rest Client call

[cols="1,1", options="header"]
|===
|Property
|Description

|`fish.payara.jaxrs.client.certificate.alias`
| The alias name of the certificate

|===

==== Example

[source, java]
----
ClientBuilder.newBuilder().property("fish.payara.jaxrs.client.certificate.alias", "someAliasName").build()

> .target(new URI("http://localhost:8080/service")).request().get(Service.class)

Reviewer: example must use https:// since functionality has to do with SSL connections.

Author: already fixed

----

> NOTE: For this to work, you need to include the certificate and the alias on the default payara keystore. In case you don't add it the default `SslContext` will be use

Reviewer 2: ```suggestion
NOTE: For this to work, you need to include the certificate with the given alias in the default payara keystore. If you don't add it the default `SslContext` will be used
```

Reviewer 2: Separately, I don't think "the default `SslContext`" actually means anything to anyone. You could probably just omit the sentence entirely.

```suggestion
NOTE: For this to work, you need to include the certificate with the given alias in the default payara keystore.
```
